dissect/target/plugins/apps/vpns/wireguard.py

```diff
@@ -0,0 +1,150 @@
+        "/opt/homebrew/etc/wireguard/*.conf",
+    ]
+
+    def __init__(self, target) -> None:
+        super().__init__(target)
+        self.configs = []
+
+    def check_compatible(self) -> bool:
+        for path in self.config_globs:
+            cfgs = list(self.target.fs.path().glob(path.lstrip("/")))
+            if len(cfgs) > 0:
+                for cfg in cfgs:
+                    self.configs.append(cfg)
+        if len(self.configs) > 0:
+            return True
+
+    @export(record=WireGuardInterfaceRecord or WireGuardPeerRecord)
+    def config(self) -> Iterator[WireGuardInterfaceRecord or WireGuardPeerRecord]:
+        """
+        Parses interface config files from wireguard installations.
+        """
+
+        for config_path in self.configs:
+
+            with self.target.fs.path(config_path).open("rt") as f:
+                config = self._parse_config(f.read())
+
+                for section in config.sections():
+
+                    if "Interface" in section:
+                        yield WireGuardInterfaceRecord(
+                            name=basename(config_path).replace(".conf", "").replace(".netdev", ""),
+                            address=config.get(section, "Address").split("/")[0],
+                            listen_port=config.get(section, "ListenPort", fallback=None),
+                            private_key=config.get(section, "PrivateKey", fallback=None),
+                            fw_mark=config.get(section, "FwMark", fallback=None),
+                            dns=config.get(section, "DNS", fallback=None),
+                            table=config.get(section, "Table", fallback=None),
+                            mtu=config.get(section, "MTU", fallback=None),
+                            preup=config.get(section, "PreUp", fallback=None),
+                            postup=config.get(section, "PostUp", fallback=None),
+                            predown=config.get(section, "PreDown", fallback=None),
+                            postdown=config.get(section, "PostDown", fallback=None),
+                            source=config_path,
+                            _target=self.target,
+                        )
+
+                    if "Peer" in section:
+                        yield WireGuardPeerRecord(
+                            name=config.get(section, "Name", fallback=None),
+                            public_key=config.get(section, "PublicKey", fallback=None),
+                            pre_shared_key=config.get(section, "PreSharedKey", fallback=None),
+                            allowed_ips=config.get(section, "AllowedIPs", fallback=None),
+                            endpoint=config.get(section, "Endpoint", fallback=None),
+                            persistent_keep_alive=config.get(section, "PersistentKeepAlive", fallback=None),
+                            source=config_path,
+                            _target=self.target,
+                        )
+
+    @internal
```

Comment:
Is there a specific reason this is `@internal`? Or would this be better as a separate function? If not, could you move this outside the class and make it a separate function (and `MultiDict` class definition).